sdk/resourcemanager/Azure.ResourceManager.Core/src/ArmClient.cs

```diff
@@ -160,5 +161,17 @@ public GenericResourceOperations GetGenericResourcesOperations(TenantResourceIde
         {
             return new GenericResourceOperations(new ClientContext(ClientOptions, Credential, BaseUri, Pipeline), id);
         }
+
+        /// <summary>
+        /// Creates a container using the lambda expression passed in.
+        /// </summary>
+        /// <typeparam name="T"> The type of container to construct. </typeparam>
+        /// <param name="func"> The lambda expression to execute. </param>
+        /// <returns> The container type requested. </returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public T GetContainer<T>(Func<ArmClientOptions, TokenCredential, Uri, HttpPipeline, T> func)
```

Comment:
Why isn't a new ClientContext passed into Func?

Reply:
Client context is internal to each RP so it cannot be transferred across library boundaries.  This keeps the object immutable inside the armClient once its created